### PR TITLE
[LoongArch64] add coreclr-libraries directory.

### PR DESCRIPTION
**Pull request by shushanhf**
[LoongArch64] add coreclr-libraries directory.

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Reflection/ImageFileMachine.cs [hunk 1]
```diff
@@ -9,5 +9,6 @@ public enum ImageFileMachine
         IA64 = 0x0200,
         AMD64 = 0x8664,
         ARM = 0x01c4,
+        LOONGARCH64 = 0xdd64,
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/Reflection/ImageFileMachine.cs:12]: This enum is deprecated. We are not adding new architectures to this enum. This change should be reverted. More details https://github.com/dotnet/runtime/issues/58970#issuecomment-917417493

File: src/libraries/System.Private.CoreLib/src/System/Reflection/ProcessorArchitecture.cs [hunk 2]
```diff
@@ -10,6 +10,7 @@ public enum ProcessorArchitecture
         X86 = 0x0002,
         IA64 = 0x0003,
         Amd64 = 0x0004,
-        Arm = 0x0005
+        Arm = 0x0005,
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/Reflection/ProcessorArchitecture.cs:13]: Dtto, this enum is deprecated. This change should be reverted.

File: src/libraries/System.Runtime.InteropServices.RuntimeInformation/tests/CheckArchitectureTests.cs [hunk 3]
```diff
@@ -40,6 +40,10 @@ public void VerifyArchitecture()
                     Assert.Equal(Architecture.S390x, processArch);
                     break;
 
+                case Architecture.Loongarch64:
+                    Assert.Equal(Architecture.Loongarch64, processArch);  //TODO:not suppoert Loongarch32 yet.
```
Comment (jkotas) [inline src/libraries/System.Runtime.InteropServices.RuntimeInformation/tests/CheckArchitectureTests.cs:44]: Delete the TODO comment

File: src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/Machine.cs [hunk 4]
```diff
@@ -129,5 +129,10 @@ public enum Machine : ushort
         /// ARM64
         /// </summary>
         Arm64 = 0xAA64,
+
+        /// <summary>
+        /// LOONGARCH64
+        /// </summary>
+        Loongarch64 = 0xDD64,
```
Comment (jkotas) [inline src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/Machine.cs:136]: The values in this enum are only relevant on Windows. I think this change should be reverted.

Notice that wasm or s390x are not in this enum either since Windows OS does not run on these architectures.

Comment (hez2010) [reply src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/Machine.cs:136]: I don't think so, see https://docs.microsoft.com/en-us/windows/win32/debug/pe-format#machine-types
`IMAGE_FILE_MACHINE_LOONGARCH64` is already a valid PE header value.

Comment (jkotas) [reply src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/Machine.cs:136]: Ok, if the PE spec defines the value, it should be fine to add it here. The value needs to match what is in the PE spec.

File: src/libraries/System.Runtime.InteropServices.RuntimeInformation/ref/System.Runtime.InteropServices.RuntimeInformation.cs [hunk 5]
```diff
@@ -14,6 +14,7 @@ public enum Architecture
         Arm64 = 3,
         Wasm = 4,
         S390x = 5,
+        Loongarch64 = 6,
```
Comment (jkotas) [inline src/libraries/System.Runtime.InteropServices.RuntimeInformation/ref/System.Runtime.InteropServices.RuntimeInformation.cs:17]: These are adding a new public APIs, so it may be a good idea to run them through a quick API review. Could you please file API proposal issue for it? https://aka.ms/api-review has details.

I am not sure about the casing. Should it be `Loongarch64` or `LoongArch64` ?

File: src/libraries/System.Private.CoreLib/src/System/Reflection/ImageFileMachine.cs (hunk 1 above)
Comment (shushanhf) [reply src/libraries/System.Private.CoreLib/src/System/Reflection/ImageFileMachine.cs:12]: OK.Thanks
I will update it later.

It's best that .NET defines a specification about the  enum for different CPUs.

File: src/libraries/System.Runtime.InteropServices.RuntimeInformation/ref/System.Runtime.InteropServices.RuntimeInformation.cs (hunk 5 above)
Comment (shushanhf) [reply src/libraries/System.Runtime.InteropServices.RuntimeInformation/ref/System.Runtime.InteropServices.RuntimeInformation.cs:17]: LoongArch64 is better, but it's not so strict.
I will update  to LoongArch64.

File: src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/Machine.cs (hunk 4 above)
Comment (shushanhf) [reply src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/Machine.cs:136]: Thanks, I will update it as the spec.

**Comment by huoyaoyuan** [comment]
Notes for updating libraries:

The reference source should also be updated together with implementation when any public api changes. For other libraries the ref source is under the `ref` folder besides the `src` directory. It can be also accessed from the corresponding Visual Studio solution. For CoreLib, the ref source is mostly under `libraries/System.Runtime/src`.

When updating the CoreCLR version of CoreLib, don't forget Mono (under `src\mono\System.Private.CoreLib`) and NativeAOT (under `src\coreclr\nativeaot\System.Private.CoreLib`, merged recently) versions.

File: src/coreclr/System.Private.CoreLib/src/System/Reflection/AssemblyName.CoreCLR.cs [hunk 6]
```diff
@@ -103,6 +103,10 @@ internal static ProcessorArchitecture CalculateProcArchIndex(PortableExecutableK
                 {
                     return ProcessorArchitecture.Arm;
                 }
+                if (ifm == ImageFileMachine.LOONGARCH64)
```
Comment (jkotas) [inline src/coreclr/System.Private.CoreLib/src/System/Reflection/AssemblyName.CoreCLR.cs:106]: This change should be reverted too.

File: src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyName.cs [hunk 7]
```diff
@@ -83,14 +83,14 @@ public ProcessorArchitecture ProcessorArchitecture
             get
             {
                 int x = (((int)_flags) & 0x70) >> 4;
-                if (x > 5)
+                if (x > 6)
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyName.cs:86]: The changes in this file should be reverted.

File: src/libraries/System.Runtime/ref/System.Runtime.cs [hunk 8]
```diff
@@ -11809,6 +11809,7 @@ public enum ImageFileMachine
         ARM = 452,
         IA64 = 512,
         AMD64 = 34404,
+        LOONGARCH64 = 25188,
```
Comment (jkotas) [inline src/libraries/System.Runtime/ref/System.Runtime.cs:11812]: The changes in this file should be reverted.

**Comment by jkotas** [comment]
I see that you are always squashing your changes into the single commit. Could you please push new commits instead of squashing? We prefer that the changes done based on codereview feedback are done as separate commits to make it easy to see what changed. We will squash all changes as part of the merge.

More details are about our PR process are in https://github.com/dotnet/runtime/blob/main/docs/pr-guide.md#quick-code-review-rules

File: src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/Machine.cs [hunk 9]
```diff
@@ -129,5 +129,10 @@ public enum Machine : ushort
         /// ARM64
         /// </summary>
         Arm64 = 0xAA64,
+
+        /// <summary>
+        /// LOONGARCH64
+        /// </summary>
+        LoongArch64 = 0x6264,
```
Comment (jkotas) [inline src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/Machine.cs:136]: It may make sense to also add `LoongArch32` to this enum since it is part of the PE file spec now. This enum should mirror PE file spec, independent on whether .NET runs on the architecture or not.

**Comment by shushanhf** [comment]
> accessed

Thanks for your review and suggestion~

File: src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/Machine.cs (hunk 9 above)
Comment (shushanhf) [reply src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/Machine.cs:136]: Thanks, I will update it.

File: src/coreclr/System.Private.CoreLib/src/System/Reflection/AssemblyName.CoreCLR.cs (hunk 6 above)
Comment (shushanhf) [reply src/coreclr/System.Private.CoreLib/src/System/Reflection/AssemblyName.CoreCLR.cs:106]: OK,Thanks.

File: src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyName.cs (hunk 7 above)
Comment (shushanhf) [reply src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyName.cs:86]: OK,Thanks.

File: src/libraries/System.Runtime/ref/System.Runtime.cs (hunk 8 above)
Comment (shushanhf) [reply src/libraries/System.Runtime/ref/System.Runtime.cs:11812]: OK,Thanks for your review~

**Comment by shushanhf** [comment]
> I see that you are always squashing your changes into the single commit. Could you please push new commits instead of squashing? We prefer that the changes done based on codereview feedback are done as separate commits to make it easy to see what changed. We will squash all changes as part of the merge.
> 
> More details are about our PR process are in https://github.com/dotnet/runtime/blob/main/docs/pr-guide.md#quick-code-review-rules

Thanks !
I will update based on the previous patch.

**Comment by shushanhf** [comment]
Now this PR can be compiled sucessfully by the CI on the native mode.

What should I do next ?

**Comment by shushanhf** [comment]
System.IO.Pipes.AccessControl -> /__w/1/s/artifacts/bin/System.IO.Pipes.AccessControl/net7.0-Debug/System.IO.Pipes.AccessControl.dll
/__w/1/s/.packages/microsoft.dotnet.apicompat/7.0.0-beta.21613.2/build/Microsoft.DotNet.ApiCompat.targets(145,5): error : MembersMustExist : Member 'public System.Reflection.PortableExecutable.Machine System.Reflection.PortableExecutable.Machine System.Reflection.PortableExecutable.Machine.LoongArch32' does not exist in the reference but it does exist in the implementation. [/__w/1/s/src/libraries/System.Reflection.Metadata/src/System.Reflection.Metadata.csproj]
##[error].packages/microsoft.dotnet.apicompat/7.0.0-beta.21613.2/build/Microsoft.DotNet.ApiCompat.targets(145,5): error : (NETCORE_ENGINEERING_TELEMETRY=Build) MembersMustExist : Member 'public System.Reflection.PortableExecutable.Machine System.Reflection.PortableExecutable.Machine System.Reflection.PortableExecutable.Machine.LoongArch32' does not exist in the reference but it does exist in the implementation.
/__w/1/s/.packages/microsoft.dotnet.apicompat/7.0.0-beta.21613.2/build/Microsoft.DotNet.ApiCompat.targets(159,5): error : MatchingRefApiCompat failed - The reference assembly doesn't match all the APIs in the implementation for '/__w/1/s/artifacts/bin/System.Reflection.Metadata/net7.0-Debug/System.Reflection.Metadata.dll'. To address either fix errors in the reference assembly (referenced as implementation in compat errors for this reverse compat check), add the issues to the baseline file '/__w/1/s/src/libraries/System.Reflection.Metadata/src/MatchingRefApiCompatBaseline.txt' or disable this check by setting RunMatchingRefApiCompat=false in this project. 


Maybe we have to update the .packages/microsoft.dotnet.apicompat for the restore error.

**Comment by huoyaoyuan** [comment]
> does not exist in the reference but it does exist in the implementation.

This means that "The reference source should also be updated together with implementation when any public api changes". And also, updating reference source for public api should require the api to be approved first. The next step is to open api proposal. I can provide some help.

File: src/libraries/System.Reflection.MetadataLoadContext/src/System.Reflection.MetadataLoadContext.csproj [hunk 10]
```diff
@@ -12,6 +12,11 @@ Commonly Used Types:
 System.Reflection.MetadataLoadContext
 System.Reflection.MetadataAssemblyResolver</PackageDescription>
   </PropertyGroup>
+
+  <PropertyGroup Condition="'$(DotNetBuildFromSource)' == 'true'">
```
Comment (jkotas) [inline src/libraries/System.Reflection.MetadataLoadContext/src/System.Reflection.MetadataLoadContext.csproj:16]: Revert this change. It is unnecessary.

File: src/libraries/System.Reflection.Metadata/ref/System.Reflection.Metadata.cs [hunk 11]
```diff
@@ -3271,6 +3271,7 @@ public enum Machine : ushort
         Amd64 = (ushort)34404,
         M32R = (ushort)36929,
         Arm64 = (ushort)43620,
+        LoongArch64 = (ushort)25188,
```
Comment (jkotas) [inline src/libraries/System.Reflection.Metadata/ref/System.Reflection.Metadata.cs:3274]: Add LoongArch32 here as well.

File: src/libraries/System.Reflection.MetadataLoadContext/tests/src/Tests/Module/ModuleTests.cs [hunk 12]
```diff
@@ -233,5 +233,23 @@ public static void GetPEKindArm()
                 Assert.Equal(ProcessorArchitecture.Arm, pa);
             }
         }
+
+        [Fact]
+        public static void GetPEKindLoongArch64()
+        {
+            using (MetadataLoadContext lc = new MetadataLoadContext(new EmptyCoreMetadataAssemblyResolver()))
+            {
+                Assembly a = lc.LoadFromByteArray(TestData.s_PlatformLoongArch64);
```
Comment (jkotas) [inline src/libraries/System.Reflection.MetadataLoadContext/tests/src/Tests/Module/ModuleTests.cs:242]: This test is testing ImageFileMachine.LOONGARCH64 that was reverted. It does not compile. Revert this change.

File: src/libraries/System.Reflection.MetadataLoadContext/src/System.Reflection.MetadataLoadContext.csproj (hunk 10 above)
Comment (shushanhf) [reply src/libraries/System.Reflection.MetadataLoadContext/src/System.Reflection.MetadataLoadContext.csproj:16]: Ok.
Thanks

File: src/libraries/System.Reflection.Metadata/ref/System.Reflection.Metadata.cs (hunk 11 above)
Comment (shushanhf) [reply src/libraries/System.Reflection.Metadata/ref/System.Reflection.Metadata.cs:3274]: OK, I will add it later.

File: src/libraries/System.Reflection.MetadataLoadContext/tests/src/Tests/Module/ModuleTests.cs (hunk 12 above)
Comment (shushanhf) [reply src/libraries/System.Reflection.MetadataLoadContext/tests/src/Tests/Module/ModuleTests.cs:242]: OK, Thanks.

File: src/libraries/System.Reflection.MetadataLoadContext/src/System/Reflection/TypeLoading/Assemblies/Ecma/EcmaAssembly.GetName.cs [hunk 13]
```diff
@@ -53,6 +53,10 @@ protected sealed override AssemblyNameData ComputeNameData()
                     data.ProcessorArchitecture = ProcessorArchitecture.Arm;
                     break;
 
+                case ImageFileMachine.LOONGARCH64:
```
Comment (jkotas) [inline src/libraries/System.Reflection.MetadataLoadContext/src/System/Reflection/TypeLoading/Assemblies/Ecma/EcmaAssembly.GetName.cs:56]: This change should be reverted too.

Comment (shushanhf) [reply src/libraries/System.Reflection.MetadataLoadContext/src/System/Reflection/TypeLoading/Assemblies/Ecma/EcmaAssembly.GetName.cs]: Thanks.
Had updated.

File: src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/PEHeaderBuilder.cs [hunk 14]
```diff
@@ -105,7 +105,7 @@ public static PEHeaderBuilder CreateLibraryHeader()
             return new PEHeaderBuilder(imageCharacteristics: Characteristics.ExecutableImage | Characteristics.Dll);
         }
 
-        internal bool Is32Bit => Machine != Machine.Amd64 && Machine != Machine.IA64 && Machine != Machine.Arm64;
+        internal bool Is32Bit => Machine != Machine.Amd64 && Machine != Machine.IA64 && Machine != Machine.Arm64 && Machine != Machine.LoongArch64;
```
Comment (jkotas) [inline src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/PEHeaderBuilder.cs:108]: This change is not really necessary. This list only needs to contain 64-bit architectures that are supported by Windows. For example, S390x is 64-bit too and it is not in this list.

Could you please revert this change?

**Comment by jkotas** [comment]
@shushanhf This API was approved so this PR is getting close to getting merged.

Could you please resolve the merge conflicts?

File: src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/PEHeaderBuilder.cs (hunk 14 above)
Comment (shushanhf) [reply src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/PEHeaderBuilder.cs:108]: OK,thanks.
I will do it now.

**Comment by shushanhf** [comment]
> @shushanhf This API was approved so this PR is getting close to getting merged.
> 
> Could you please resolve the merge conflicts?

Is this PR ok now ?

**Comment by jkotas** [comment]
> Is this PR ok now ?

No, it has a bunch of conflicts. You should see "This branch has conflicts that must be resolved".

You need to merge from current main to get these conflicts resolved.

**Comment by shushanhf** [comment]
> > Is this PR ok now ?
> 
> No, it has a bunch of conflicts. You should see "This branch has conflicts that must be resolved".
> 
> You need to merge from current main to get these conflicts resolved.

I have a question that needs your help:
How to rebase the PR's patch to the latest main branch ?

**Comment by shushanhf** [comment]
> > > Is this PR ok now ?
> > 
> > 
> > No, it has a bunch of conflicts. You should see "This branch has conflicts that must be resolved".
> > You need to merge from current main to get these conflicts resolved.
> 
> I have a question that needs your help: How to rebase the PR's patch to the latest main branch ?

If I merge the main directly to this PR and resolv the conflicts, then I push to this PR, is it OK?

**Comment by jkotas** [comment]
Yep

**Comment by shushanhf** [comment]
> Yep

OK,  thanks.
I will update now.

**Comment by shushanhf** [comment]
> Yep

OK,  thanks.
I will update now.

> Yep

Should this file be modified ？


```
diff --git a/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/RuntimeInformation.ProcessArchitecture.cs b/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/RuntimeInformation.ProcessArchitecture.cs
index 298d039..0da595b 100644
--- a/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/RuntimeInformation.ProcessArchitecture.cs
+++ b/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/RuntimeInformation.ProcessArchitecture.cs
@@ -18,6 +18,8 @@ public static Architecture ProcessArchitecture
             => Architecture.Wasm;
 #elif TARGET_S390X
             => Architecture.S390x;
+#elif TARGET_LOONGARCH64
+            => Architecture.LoongArch64;
 #else
 #error Unknown Architecture
 #endif
```

File: src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/ManagedTextSection.cs [hunk 15]
```diff
@@ -90,7 +90,7 @@ internal sealed class ManagedTextSection
         /// If set, the module contains instructions that assume a 64 bit instruction set. For example it may depend on an address being 64 bits.
         /// This may be true even if the module contains only IL instructions because of PlatformInvoke and COM interop.
         /// </summary>
-        internal bool Requires64bits => Machine == Machine.Amd64 || Machine == Machine.IA64 || Machine == Machine.Arm64;
+        internal bool Requires64bits => Machine == Machine.Amd64 || Machine == Machine.IA64 || Machine == Machine.Arm64 || Machine == Machine.LoongArch64;
```
Comment (jkotas) [inline src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/ManagedTextSection.cs:93]:
```suggestion
        internal bool Requires64bits => Machine == Machine.Amd64 || Machine == Machine.IA64 || Machine == Machine.Arm64;
```
No needed for non-Windows ports

File: src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Architecture.cs [hunk 16]
```diff
@@ -10,6 +10,7 @@ public enum Architecture
         Arm,
         Arm64,
         Wasm,
-        S390x
+        S390x,
+        LoongArch64
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Architecture.cs:14]: Could you please also add this one to `src/libraries/System.Runtime/ref/System.Runtime.cs`? It will fix one of the build breaks that you are seeing.

Comment (shushanhf) [reply src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Architecture.cs:14]: OK, thanks

File: src/coreclr/System.Private.CoreLib/System.Private.CoreLib.csproj [hunk 17]
```diff
@@ -78,6 +78,10 @@
     <PlatformTarget>AnyCPU</PlatformTarget>
     <DefineConstants>$(DefineConstants);TARGET_ARM64</DefineConstants>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(Platform)' == 'loongarch64'">
```
Comment (jkotas) [inline src/coreclr/System.Private.CoreLib/System.Private.CoreLib.csproj:81]: Could you please also add `'$(Platform)' == 'loongarch64'` for `Is64Bit` in `src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems`?

Comment (shushanhf) [reply src/coreclr/System.Private.CoreLib/System.Private.CoreLib.csproj:81]: Thanks for your review.
I will do it right now.

**Comment by jkotas** [comment]
The last build break is: "The generated CompatibilityMap differs from runtime.compatibility.json and UpdateRuntimeFiles was not specified.  Please specify UpdateRuntimeFiles=true to commit the changes."

Could you please run the build with `/p:UpdateRuntimeFiles=true` and commit the result? It should fix it.

**Comment by shushanhf** [comment]
> The last build break is: "The generated CompatibilityMap differs from runtime.compatibility.json and UpdateRuntimeFiles was not specified. Please specify UpdateRuntimeFiles=true to commit the changes."
> 
> Could you please run the build with `/p:UpdateRuntimeFiles=true` and commit the result? It should fix it.

Thanaks ~
I will do it now.

**Comment by shushanhf** [comment]
> The last build break is: "The generated CompatibilityMap differs from runtime.compatibility.json and UpdateRuntimeFiles was not specified. Please specify UpdateRuntimeFiles=true to commit the changes."
> 
> Could you please run the build with `/p:UpdateRuntimeFiles=true` and commit the result? It should fix it.

Hi,  @jkotas 

Running the command `./build.sh libs  -lc release`  with the following patch,
```
qiao@deb-x64:~/work_qiao/runtime$ git diff
diff --git a/src/libraries/Microsoft.NETCore.Platforms/src/Microsoft.NETCore.Platforms.csproj b/src/libraries/Microsoft.NETCore.Platforms/src/Microsoft.NETCore.Platforms.csproj
index 1351a67f30f..5c3e395a0fc 100644
--- a/src/libraries/Microsoft.NETCore.Platforms/src/Microsoft.NETCore.Platforms.csproj
+++ b/src/libraries/Microsoft.NETCore.Platforms/src/Microsoft.NETCore.Platforms.csproj
@@ -75,7 +75,7 @@
                           RuntimeJson="runtime.json"
                           CompatibilityMap="runtime.compatibility.json"
                           RuntimeDirectedGraph="$(OutputPath)runtime.json.dgml"
-                          UpdateRuntimeFiles="$(UpdateRuntimeFiles)" />
+                          UpdateRuntimeFiles="True" />
   </Target>
 
 </Project>
```

Is this right ?

I didn't meet the error https://github.com/dotnet/runtime/pull/62888/files#annotation_2513131987
by the command `./build.sh libs  -lc release`.
How to test and reproduce the error ?
Thanks

**Comment by jkotas** [comment]
I have run:
```
cd src/libraries/Microsoft.NETCore.Platforms/src
dotnet build -c Release Microsoft.NETCore.Platforms.csproj /p:UpdateRuntimeFiles=true /t:UpdateRuntimeJson
```
and pushed up the result. It should fix the build break.

**Comment by shushanhf** [comment]
> I have run:
> 
> ```
> cd src/libraries/Microsoft.NETCore.Platforms/src
> dotnet build -c Release Microsoft.NETCore.Platforms.csproj /p:UpdateRuntimeFiles=true /t:UpdateRuntimeJson
> ```
> 
> and pushed up the result. It should fix the build break.

Thanks,

**Comment by shushanhf** [comment]
> build -c Release Microsoft.NETCore.Platforms.csproj /p:UpdateRuntimeFiles=true /t:UpdateRuntimeJson

@jkotas 
Thanks for you!

I runned the command and the results is :

```
qiao@deb-x64:~/work_qiao/runtime/src/libraries/Microsoft.NETCore.Platforms/src$ ~/work_qiao/runtime/.dotnet/dotnet build -c Release Microsoft.NETCore.Platforms.csproj /p:UpdateRuntimeFiles=true /t:UpdateRuntimeJson
Microsoft (R) Build Engine version 17.0.0+c9eb9dd64 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  Restored /data2_users/work_qiao/runtime/src/libraries/Microsoft.NETCore.Platforms/src/Microsoft.NETCore.Platforms.csproj (in 217 ms).

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:04.42
```

**Comment by jkotas** [comment]
Closing&reopening to trigger the CI

**Review by jkotas — APPROVED** [review]
